addons/l10n_sg_ubl_pint/models/account_move.py

```diff
@@ -0,0 +1,25 @@
+import uuid
+
+from odoo import fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    l10n_sg_ubl_pint_uuid = fields.Char(string="Document UUID (SG)", copy=False, help="Universally unique identifier of the Invoice")
+
+    def _l10n_sg_get_uuid(self):
+        """ The rule is only to provide a uuid. To keep it as simple as possible:
+        - We will only set that uuid when required (generating the file).
+        - We reset it when the invoice is reset to draft.
+        """
+        self.ensure_one()
+        if not self.l10n_sg_ubl_pint_uuid:
+            self.l10n_sg_ubl_pint_uuid = uuid.uuid4()
+        return self.l10n_sg_ubl_pint_uuid
```

Comment:
Well, the idea is to avoid storing a uuid in the database if not necessary.
Using a default value means every single invoice will have a uuid, even though you may not need it on all of them.

Reply:
What about doing something equivalent to `_l10n_de_datev_get_guid` ? Apparently this function was suggested by AL to avoids storing an additional field, but it is still deterministic. What do you think ?

Reply:
Hm looks like a good idea yes, we can use that system